routers/api/v1/api.go

```diff
@@ -807,6 +807,13 @@ func Routes() *web.Route {
 								Get(repo.GetIssueCommentReactions).
 								Post(reqToken(), bind(api.EditReactionOption{}), repo.PostIssueCommentReaction).
 								Delete(reqToken(), bind(api.EditReactionOption{}), repo.DeleteIssueCommentReaction)
+							m.Combo("/assets").Get(repo.ListIssueCommentAttachments).
+								Post(reqToken(), reqRepoWriter(models.UnitTypeIssues), repo.CreateIssueCommentAttachment)
+						})
+						m.Group("/assets", func() {
+							m.Combo("/{asset}").Get(repo.GetIssueCommentAttachment).
+								Patch(reqToken(), reqRepoWriter(models.UnitTypeIssues), bind(api.EditAttachmentOptions{}), repo.EditIssueCommentAttachment).
+								Delete(reqToken(), reqRepoWriter(models.UnitTypeIssues), repo.DeleteIssueCommentAttachment)
```

Comment:
these permissions checks are incorrect.

Reply:
Yeah, this is weird.
You apply two methods for checking permissions: Once here via middleware, but it seems you just copy pasted these, and then in certain handlers (eg `GetIssueAttachment()`) too.
I'd factor this in a way that you have an explicit permission check in middleware (you'll need at least two: reqReadAttachment, reqWriteAttachment), and no need for permission checks in the route handlers.